grails-app/services/org/pih/warehouse/data/ProductSupplierDataService.groovy

```diff
@@ -302,7 +302,8 @@ class ProductSupplierDataService {
                 'manufacturerCode': manufacturerCode,
                 'supplierCode': supplierCode,
         ])
-        def productSupplier = data ? ProductSupplier.get(data.first().id) : null
+        ArrayList<ProductSupplier> productSuppliers = data?.collect{ it -> ProductSupplier.get(it?.id) }?.sort{ !it?.active }
+        def productSupplier = !!productSuppliers ? productSuppliers?.first() : null
```

Comment:
I don't think Groovy supports double exclamation. It'll handle the boolean without this. What you could say is 

```
!productSuppliers.empty ? productSuppliers.first() : null
```

Reply:
It actually does - @drodzewicz showed me "the magic" yesterday as he also tried to use it. It was helpful to shorten something like this: `productSuppliers.size() > 0 ? ... : ...`, because if a list is empty, `!!productSuppliers` returns false, and if it's not empty, it returns true, so it does work like `productSuppliers.size() > 0` but if you don't like this idea, I'll proceed with what you've written.